chalice/app.py

```diff
@@ -155,6 +199,16 @@ def maybe_fake_504() -> typing.Optional[chalice.Response]:
         res_headers = dict(flask_res.headers)
         # API Gateway/Cloudfront adds a duplicate Content-Length with a different value (not sure why)
         res_headers.pop("Content-Length", None)
+
+        # observability logic is sent to a different thread so as not to delay response
+        thread = threading.Thread(
+            name='observables',
+            target=observe_request,
+            args=(path, app.current_request.method, flask_res._status_code)
+        )
+        thread.setDaemon(False)  # not a daemon, function returns, but program waits until metrics are sent
```

Comment:
Doesn't this mean that we might _not_ finish sending the logging data?

Reply:
I don't think (!) it matters. Looking at AWS' bootstrap.py there is just an infinite loop that repeatedly calls the handler for every incoming request. My intuition is that the process will be killed at the 5min deadline irregardless of whether there are any threads, daemons or not.

IOW, merely the fact that this is happening in a thread makes it more likely to not be finished.

https://docs.aws.amazon.com/lambda/latest/dg/running-lambda-code.html

> Background processes or callbacks initiated by your Lambda function that did not complete when the function ended resume if AWS Lambda chooses to reuse the Execution Context. You should make sure any background processes or callbacks (in case of Node.js) in your code are complete before the code exits.

Reply:
We could have a little heuristic that performs this synchronously if we are within 30s of the 5min deadline and logs a warning if we are within 5s of the deadline because that close to the deadline even a synchronous execution is likely going to be interrupted.